src/features/counter/users/userSlice.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../../../types/User';
+import { getUsers } from '../../../api/users';
+
+type UsersState = {
+  users: User[];
+  selectedUser: User | null;
+  loading: boolean;
+  error: string;
+};
+
+const initialState: UsersState = {
+  users: [],
+  selectedUser: null,
+  loading: false,
+  error: '',
+};
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {
+    addSelectedUser: (state, action: PayloadAction<User>) => {
+      state.selectedUser = action.payload;
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(init.pending, state => {
+      state.loading = true;
+    });
+
+    builder.addCase(init.fulfilled, (state, action) => {
+      state.users = action.payload;
+      state.loading = false;
+    });
+
+    builder.addCase(init.rejected, state => {
+      state.error = 'Something went wrong!';
+      state.loading = false;
+    });
+  },
+});
+
+export default usersSlice;
```

Comment:
The export should be `export default usersSlice.reducer;` to correctly export the reducer function for use in the store.